scripts/workers/evaluator.js

```diff
@@ -0,0 +1,36 @@
+"use strict";
+
+// Initialize some global variable to use directly inside eval()
+const abs = Math.abs;
+const floor = Math.floor;
+const ceil = Math.ceil;
+
+const sin = Math.sin;
+const cos = Math.cos;
+const tan = Math.tan;
+
+const log = Math.log;
+
+// Handle message events from main script.
+globalThis.addEventListener("message", function(event) {
+    try {
+        let result = eval(event.data);
+
+        // Handle divide by 0
+        if (Math.abs(result) === Infinity) {
+            throw new Error("This operation is not allowed");
+        }
```

Comment:
Can we use the `isFinite` method instead of checking in this `Math.abs(result) === Infinity`?
```suggestion
        if (!isFinite(result)) {
            throw new Error("This operation is not allowed");
        }
```

Reply:
Used the `isFinite` function. 
https://github.com/dhruvdhaduk-simform/scientific-calc/pull/3/commits/df6736e08c3a1b05886a175813c7f6a17458da7f